shared/js/project.js

```diff
@@ -190,7 +200,17 @@ module.exports = {
 
     current = data;
     if (isNewChannel) {
-      location.href = current.level + '#' + current.id + '/edit';
+      // We have a new channel, meaning either we had no channel before, or
+      // we've changed channels.
+
+      if (location.hash || !window.history.pushState) {
+        // We're using a hash route or don't support replace state. Use our hash
+        // based route to ensure we don't have a page load.
+        location.href = current.level + '#' + current.id + '/edit';
+      } else {
+        window.history.pushState(null, document.title,
```

Comment:
I'm also realizing we may want a reload here, at least in some cases. For example, if I'm on a share page and click remix I want to (a) hit the server to make sure I'm logged in and (b) want hideSource to be false.